UseCases/Sources/ManualAsyncHTTPClient/main.swift

```diff
@@ -0,0 +1,109 @@
+import AsyncHTTPClient
+import ContextPropagation
+import NIOHTTP1
+
+let server = FakeHTTPServer(
+    instrumentationMiddlewares: [FakeTracer.Middleware(tracer: FakeTracer())]
+) { context, request, client -> FakeHTTPResponse in
+    print("=== Perform subsequent request ===")
+    let outgoingRequest = try! HTTPClient.Request(
+        url: "https://swift.org",
+        headers: ["Accept": "application/json"]
+    )
+    client.execute(context, request: outgoingRequest)
+//    client.execute(request: outgoingRequest, inContext: context)
+    return FakeHTTPResponse()
+}
+
+print("=== Receive HTTP request on server ===")
+server.receive(try! HTTPClient.Request(url: "https://swift.org"))
+
+// MARK: - InstrumentingHTTPClient
+
+struct InstrumentingHTTPClient {
+    private let client = HTTPClient(eventLoopGroupProvider: .createNew)
+    private let instrumentationMiddlewares: [InstrumentationMiddleware<HTTPHeaders, HTTPHeaders>]
```

Comment:
Is the  `[]` getting a bit annoying perhaps?

Should we write an `InstrumentationMiddleware` which is an aggregate or `MultiplexingInstrumentationMiddleware` that can store the array and in all cutpoints we simply call "the middleware" (name to be bikeshed) rather than having to implement the array traversal many times?

You can have a look at https://github.com/apple/swift-log/blob/87c2553da204fc269be89aec39e38217d65c4980/Sources/Logging/Logging.swift#L459-L520 to see how we do it in logging

Reply:
Sounds good! 👍 I like the multiplex approach, will take a look. I think that's also not blocked by the naming as we'll always have multiple "things" to be called.

Reply:
Yup, just something to do separately at some point -- just because  the `[]` passing around gets annoying :)

Reply:
Reported in #14